src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from 'react';
+
+import * as RadixAvatar from '@radix-ui/react-avatar';
+import { IconCurrencyEthereum } from '../Icons';
+
+import { AVATAR_ICON_SIZE } from './AvatarIconSize.constants';
+import styles from './Avatar.module.scss';
+
+export interface AvatarProps {
+  imageURL?: string;
+  size: 'extra small' | 'small' | 'regular' | 'medium';
+  type: 'circle' | 'square';
+  userFriendlyName?: string;
+  badgeContent?: string;
+  statusType?: 'active' | 'idle' | 'busy' | 'offline' | 'unread';
+}
+
+export const Avatar = ({
+  type = 'circle',
+  size = 'regular',
+  imageURL,
+  userFriendlyName,
+  statusType,
+  badgeContent
+}: AvatarProps) => {
+  const [avatarName, setAvatarName] = useState('');
+
+  // Used to fetch the first two letter of the name
+  useEffect(() => {
+    if (userFriendlyName) {
+      const userNameArray = userFriendlyName.split(' ');
+      let charFromName = userNameArray[0].substring(0, 1);
+      charFromName += userNameArray.length >= 2 ? userNameArray[1].substring(0, 1) : userNameArray[0].substring(1, 2);
+      setAvatarName(charFromName);
+    }
+  }, [userFriendlyName]);
```

Comment:
Move this functionality out to a utility function in `Avatar.tsx`, i.e. `getInitials`, then call the function in the component body? You won't need to save state, just something like `const initials = userFriendlyName  && getInitials(userFriendlyName)`.

I understand using `useEffect` so the computation doesn't run on every render, but `useEffect` comes with its own overheads. If the function you're running as a side-effect isn't very resource intensive (like this one), it's best to just calculate it on render as it makes the component easier to maintain. In fact, it's generally advised to avoid using `useEffect` unless you're syncing with something outside of React - I think a `useMemo` would be more appropriate here if you were needing to save resources.

[There's a little write up here](https://beta.reactjs.org/learn/you-might-not-need-an-effect) that covers this more in-depth!

Reply:
Thanks for the info. I updated it to `const initials = userFriendlyName  && getInitials(userFriendlyName)`